### PR TITLE
Several fixes from PHPStan

### PR DESCRIPTION
**Pull request by carusogabriel**
I've made a several clean up in our `phpstan.neon` file by fixing some phpDocs, asserting not nullable values, rewrite some conditions, etc :)

### CODE REVIEW

File: src/Rector/Contrib/Nette/DI/CompilerCompileArgumentsRector.php [hunk 1]
```diff
@@ -38,13 +38,15 @@ public function __construct(MethodCallAnalyzer $methodCallAnalyzer, IdentifierRe
         $this->identifierRenamer = $identifierRenamer;
     }
 
+    /**
+     * @param MethodCall $node
+     */
```
Comment (TomasVotruba) [inline src/Rector/Contrib/Nette/DI/CompilerCompileArgumentsRector.php]: Good ones :+1: United with other comments

File: packages/NodeTypeResolver/src/TypeContext.php [hunk 2]
```diff
@@ -78,13 +78,12 @@ public function addVariableWithTypes(string $variableName, array $variableTypes)
         $this->variableTypes[$variableName] = $variableTypes;
     }
 
-    public function enterClassLike(ClassLike $classLikeNode): void
+    public function enterClassLike(Class_ $classLikeNode): void
```
Comment (TomasVotruba) [inline packages/NodeTypeResolver/src/TypeContext.php]: This should also accept interface and traits as well. Maybe we could add test case to cover it?

File: src/Rector/Contrib/PhpParser/IdentifierRector.php [hunk 3]
```diff
@@ -68,16 +67,17 @@ public function __construct(
         $this->nodeTypeResolver = $nodeTypeResolver;
     }
 
+    /**
+     * @param PropertyFetch $node
+     */
     public function isCandidate(Node $node): bool
```
Comment (TomasVotruba) [inline src/Rector/Contrib/PhpParser/IdentifierRector.php]: I see now that it's isCandidate() method. It accepts any node so the annotations here is miss leading

File: packages/ReflectionDocBlock/tests/NodeAnalyzer/DocBlockAnalyzerTest.php [hunk 4]
```diff
@@ -30,6 +30,8 @@ public function testHasAnnotation(): void
     public function testRemoveAnnotationFromNode(): void
     {
         $node = $this->createNodeWithDoc('@param ParamType $paramName');
+        assert($node->getDocComment() !== null);
+
```
Comment (TomasVotruba) [inline packages/ReflectionDocBlock/tests/NodeAnalyzer/DocBlockAnalyzerTest.php]: Why not PHPUnit assert?

File: src/Rector/Contrib/PHPUnit/SpecificMethod/AssertPropertyExistsRector.php [hunk 5]
```diff
@@ -74,11 +76,13 @@ public function isCandidate(Node $node): bool
     }
 
     /**
-     * @var MethodCall
+     * @param MethodCall $methodCallNode
      */
```
Comment (TomasVotruba) [inline src/Rector/Contrib/PHPUnit/SpecificMethod/AssertPropertyExistsRector.php]: Here it's correct

File: packages/ReflectionDocBlock/tests/NodeAnalyzer/DocBlockAnalyzerTest.php (hunk 4 above)
Comment (carusogabriel) [reply packages/ReflectionDocBlock/tests/NodeAnalyzer/DocBlockAnalyzerTest.php]: We can add the `assertNotNull`, but this `assert` is needed it to fix PHPStan :(

Comment (TomasVotruba) [reply packages/ReflectionDocBlock/tests/NodeAnalyzer/DocBlockAnalyzerTest.php]: Well, the method above only returns node with doc comment, so the code here is correct.

That's fail positive by PHPStan. It should not pollute code just to make it silent on places where it can't detect correctly. And it's completely fine, I don't assume to know it all.
That's what `parameters > ignoreErrors` are for.

File: easy-coding-standard.neon [hunk 6]
```diff
@@ -10,7 +10,7 @@ checkers:
     PHP_CodeSniffer\Standards\Generic\Sniffs\Files\LineLengthSniff:
         absoluteLineLimit: 120
     PHP_CodeSniffer\Standards\Generic\Sniffs\Metrics\CyclomaticComplexitySniff:
-        absoluteComplexity: 5
+        absoluteComplexity: 6
```
[the remark this answers is not in the document]
Comment (carusogabriel) [reply easy-coding-standard.neon]: I just bump it to take care of it in a next CS PR. I want to propose some stuff :)

File: packages/ReflectionDocBlock/tests/NodeAnalyzer/DocBlockAnalyzerTest.php (hunk 4 above)
Comment (carusogabriel) [reply packages/ReflectionDocBlock/tests/NodeAnalyzer/DocBlockAnalyzerTest.php]: Yes, I just remove some ignored error as we have 100+, and IMHO isn't healthy to our code. I'll try to remove this `assert`.

File: src/Rector/Contrib/PhpParser/IdentifierRector.php (hunk 3 above)
Comment (carusogabriel) [reply src/Rector/Contrib/PhpParser/IdentifierRector.php]: Just noticed that as well. Gonna change to an internal `@var`

File: packages/NodeTypeResolver/src/TypeContext.php (hunk 2 above)
Comment (carusogabriel) [reply packages/NodeTypeResolver/src/TypeContext.php]: Gonna add this tests than

File: easy-coding-standard.neon (hunk 6 above)
[the remark this answers is not in the document]
Comment (TomasVotruba) [reply easy-coding-standard.neon]: I'm afraid we could forget it, but ok :)

[the remark this answers is not in the document]
Comment (carusogabriel) [reply easy-coding-standard.neon]: Don't worry, I won't forget :smile:

File: packages/ReflectionDocBlock/tests/NodeAnalyzer/DocBlockAnalyzerTest.php (hunk 4 above)
Comment (carusogabriel) [reply packages/ReflectionDocBlock/tests/NodeAnalyzer/DocBlockAnalyzerTest.php]: > Well, the method above only returns node with doc comment, so the code here is correct.

Actually, it can return `null`, just checked: https://github.com/nikic/PHP-Parser/blob/master/lib/PhpParser/NodeAbstract.php#L115

I'm pushing a new commit with `assertNotNull`, but we must use `assert` here. Otherwise, is a test class, is good to do it :)

Comment (TomasVotruba) [reply packages/ReflectionDocBlock/tests/NodeAnalyzer/DocBlockAnalyzerTest.php]: I mean the code line above:

```php
$this->createNodeWithDoc('@param ParamType $paramName');
```

Comment (TomasVotruba) [reply packages/ReflectionDocBlock/tests/NodeAnalyzer/DocBlockAnalyzerTest.php]: > Yes, I just remove some ignored error as we have 100+, and IMHO isn't healthy to our code. 

Why do you think number of ignore errors is related to code quality?
Well, it should be called "ignored reports", since it's often PHPStan errors.

File: packages/NodeTypeResolver/src/TypeContext.php (hunk 2 above)
Comment (carusogabriel) [reply packages/NodeTypeResolver/src/TypeContext.php]: I'll rollback this one for now and fix it in a different PR with tests :)

**Comment by carusogabriel** [comment]
@TomasVotruba Should be ready to merge, thanks for the review :+1:

File: src/Rector/Contrib/PHPUnit/SpecificMethod/AssertPropertyExistsRector.php [hunk 7]
```diff
@@ -59,7 +59,7 @@ public function isCandidate(Node $node): bool
             return false;
         }
 
-        /** @var MethodCall $methodCallAnalyzer */
+        /** @var MethodCall $methodCallNode */
         $methodCallNode = $node;
```
Comment (TomasVotruba) [inline src/Rector/Contrib/PHPUnit/SpecificMethod/AssertPropertyExistsRector.php]: Nice catch!

File: src/Rector/Contrib/PHPUnit/SpecificMethod/AssertPropertyExistsRector.php (hunk 5 above)
Comment (TomasVotruba) [reply src/Rector/Contrib/PHPUnit/SpecificMethod/AssertPropertyExistsRector.php]: Do you know about Sniff/Fixer that could check that?

File: src/Rector/Contrib/Symfony/Controller/AddFlashRector.php [hunk 8]
```diff
@@ -54,6 +55,9 @@ public function isCandidate(Node $node): bool
         return true;
     }
 
+    /**
+     * @param MethodCall $node
+     */
     public function refactor(Node $node): ?Node
```
Comment (TomasVotruba) [inline src/Rector/Contrib/Symfony/Controller/AddFlashRector.php]: Nice :+1:

File: src/Rector/Contrib/Symfony/Controller/RedirectToRouteRector.php [hunk 9]
```diff
@@ -74,10 +81,13 @@ public function refactor(Node $node): ?Node
     }
 
     /**
+     * @param MethodCall $node
+     *
      * @return mixed[]
      */
     private function resolveArguments(Node $node): array
```
Comment (TomasVotruba) [inline src/Rector/Contrib/Symfony/Controller/RedirectToRouteRector.php]: I think here it could be typehint instead of `Node`, right?

File: src/Rector/Contrib/Symfony/DependencyInjection/ContainerBuilderCompileEnvArgumentRector.php [hunk 10]
```diff
@@ -28,18 +28,18 @@ public function __construct(MethodCallAnalyzer $methodCallAnalyzer, NodeFactory
...
     {
-        $isMethodCall = $this->methodCallAnalyzer->isTypeAndMethods(
+        if ($this->methodCallAnalyzer->isTypeAndMethods(
             $node,
             'Symfony\Component\DependencyInjection\ContainerBuilder',
             ['compile']
-        );
-
-        if ($isMethodCall === false) {
+        ) === false) {
```
Comment (TomasVotruba) [inline src/Rector/Contrib/Symfony/DependencyInjection/ContainerBuilderCompileEnvArgumentRector.php]: Why this change?

There is too much going on, resolving, validation and comparing to `false`. It's much less readable then before

File: src/Rector/Contrib/PHPUnit/SpecificMethod/AssertPropertyExistsRector.php (hunk 5 above)
Comment (carusogabriel) [reply src/Rector/Contrib/PHPUnit/SpecificMethod/AssertPropertyExistsRector.php]: I'll search for, but AFAIK: no :(

File: packages/ReflectionDocBlock/tests/NodeAnalyzer/DocBlockAnalyzerTest.php [hunk 11]
```diff
@@ -30,6 +30,9 @@ public function testHasAnnotation(): void
     public function testRemoveAnnotationFromNode(): void
     {
         $node = $this->createNodeWithDoc('@param ParamType $paramName');
+
+        assert($node->getDocComment() !== null);
+        $this->assertNotNull($node->getDocComment());
```
[the remark this answers is not in the document]
Comment (carusogabriel) [reply packages/ReflectionDocBlock/tests/NodeAnalyzer/DocBlockAnalyzerTest.php]: No my friend, remember https://github.com/nikic/PHP-Parser/blob/master/lib/PhpParser/NodeAbstract.php#L115?

But if you really judge it necessary, I can remove it.

File: src/Rector/Contrib/Symfony/Controller/RedirectToRouteRector.php (hunk 9 above)
Comment (carusogabriel) [reply src/Rector/Contrib/Symfony/Controller/RedirectToRouteRector.php]: Right, thanks for the catch, gonna search for similars :+1:

File: src/Rector/Contrib/Symfony/DependencyInjection/ContainerBuilderCompileEnvArgumentRector.php (hunk 10 above)
Comment (carusogabriel) [reply src/Rector/Contrib/Symfony/DependencyInjection/ContainerBuilderCompileEnvArgumentRector.php]: We do this everywhere, for example: https://github.com/rectorphp/rector/blob/master/src/Rector/Contrib/Symfony/DependencyInjection/CompilerAddPassPriorityRector.php#L42

File: packages/ReflectionDocBlock/tests/NodeAnalyzer/DocBlockAnalyzerTest.php (hunk 11 above)
[the remark this answers is not in the document]
Comment (TomasVotruba) [reply packages/ReflectionDocBlock/tests/NodeAnalyzer/DocBlockAnalyzerTest.php]: Nono, that's not what I mean. 
See the `createNodeWithDoc()`

File: src/Rector/Contrib/Symfony/DependencyInjection/ContainerBuilderCompileEnvArgumentRector.php (hunk 10 above)
Comment (TomasVotruba) [reply src/Rector/Contrib/Symfony/DependencyInjection/ContainerBuilderCompileEnvArgumentRector.php]: I see, ok. I'll try to upgrade my cyclomatix complecity for reading :)

File: phpstan.neon [hunk 12]
```diff
@@ -1,25 +1,18 @@
 parameters:
     ignoreErrors:
         # false positive, has annotation type above (@todo recheck for possible ignored positives)
-        - '#Access to an undefined property PhpParser\\Node::\$args#' # 20
-        - '#Access to an undefined property PhpParser\\Node::\$var#' # 2
-        - '#Access to an undefined property PhpParser\\Node::\$name#' # 12
+        - '#Access to an undefined property PhpParser\\Node::\$name#' # 4
```
Comment (TomasVotruba) [inline phpstan.neon]: Btw, what the number means? Could you add explanation to first item?

Comment (carusogabriel) [reply phpstan.neon]: Are the number of occurrences, so we can give priority to those ones that have more. I'll add it

File: packages/ReflectionDocBlock/tests/NodeAnalyzer/DocBlockAnalyzerTest.php (hunk 11 above)
[the remark this answers is not in the document]
Comment (carusogabriel) [reply packages/ReflectionDocBlock/tests/NodeAnalyzer/DocBlockAnalyzerTest.php]: Oh, perfect, now I understand your complaint :smile: 

I'll try to fix it, or ignore it so we can merge it :+1:

**Comment by carusogabriel** [comment]
@TomasVotruba Done, should be RTM :tada:

File: src/Rector/Architecture/RepositoryAsService/ReplaceParentRepositoryCallsByRepositoryPropertyRector.php [hunk 13]
```diff
@@ -59,6 +59,8 @@ private function getEntityRepositoryPublicMethodNames(): array
     {
         $entityRepositoryReflection = $this->smartClassReflector->reflect('Doctrine\ORM\EntityRepository');
 
-        return array_keys($entityRepositoryReflection->getImmediateMethods());
+        if ($entityRepositoryReflection !== null) {
+            return array_keys($entityRepositoryReflection->getImmediateMethods());
+        }
     }
```
[the remark this answers is not in the document]
Comment (TomasVotruba) [reply src/Rector/Architecture/RepositoryAsService/ReplaceParentRepositoryCallsByRepositoryPropertyRector.php]: Could you check this? There should be `[]` on return always, but here it's `void`

That should be reported by PHPStan

[the remark this answers is not in the document]
Comment (carusogabriel) [reply src/Rector/Architecture/RepositoryAsService/ReplaceParentRepositoryCallsByRepositoryPropertyRector.php]: Oh, missed that one, thanks
